### ISSUE TITLE
`exception Apron.Manager.Error  {  exn = Exc_invalid_argument; funid = Funid_substitute_texpr_array;   msg = unknown variable d#311#arg in the environment; }`

### ISSUE THREAD

**Issue by michael-schwarz** [goblint/analyzer#1249]
The test `ldv-linux-3.16-rc1/205_9a_array_safes_linux-3.16-rc1.tar.xz-205_9a-drivers--net--wireless--ti--wlcore--wlcore_spi.ko-entry_point.cil.out.i` crashes in our sv-comp runs with error

```
Fatal error: exception Apron.Manager.Error
 {  exn = Exc_invalid_argument; funid = Funid_substitute_texpr_array;
  msg = unknown variable d#311#arg in the environment; }
  ```

An extracted minimal example looks like this: Of course, this program here has UB because `a` is a null-pointer, but the analyses should nevertheless run and report an error and not just crash.

~~~C
// SKIP PARAM: --set ana.activated[+] apron
int *a;
int b;
void c(int d) {
    // *a is a null pointer here, so we should warn but maybe not crash
    *a = d;
}
int main() {
    c(b);
}
~~~

This is not highest priority, but might be worth having a look at, since it is a strange bug.

**Comment by michael-schwarz** [comment]
According to the stack trace the issue seems to occur in `combine_env`.

```
Raised by primitive operation at Abstract1.substitute_texpr_with in file "abstract1.ml", line 291, characters 0-58
Called from Goblint_lib__ApronDomain.AOpsPureOfImperative.substitute_exp in file "src/cdomains/apron/apronDomain.apron.ml", line 203, characters 4-36
Called from Goblint_lib__RelationAnalysis.SpecFunctor.assign_from_globals_wrapper in file "src/analyses/apron/relationAnalysis.apron.ml", line 125, characters 15-24
Called from Stdlib__List.fold_left in file "list.ml", line 121, characters 24-34
Called from Goblint_lib__RelationAnalysis.SpecFunctor.combine_env in file "src/analyses/apron/relationAnalysis.apron.ml", line 399, characters 22-322
Called from Goblint_lib__MCP.MCP2.combine_env.f in file "src/analyses/mCP.ml", line 533, characters 17-79
```

**Comment by sim642** [comment]
I recently fixed some tiny issue where a `varinfo_tracked` check was missing but I'll have to see where I did that.
